### ISSUE TITLE
App Store rejection for not including NSPhotoLibraryUsageDescription

### ISSUE THREAD

**Issue by brblakley** [mikebuss/MTBBarcodeScanner#86]
I created a build of my app with the iOS 10 SDK and submitted it to iTunes Connect and got the following rejection a few minutes later in an automated email:

> This app attempts to access privacy-sensitive data without a usage description. The app's Info.plist must contain an NSPhotoLibraryUsageDescription key with a string value explaining to the user how the app uses this data.

I was able to track down the cause of the rejection to this line:

```
return [UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera];
```

https://github.com/mikebuss/MTBBarcodeScanner/blob/daab12a707ffeed5524a1fc7da1be94e7b01fe00/Classes/ios/Scanners/MTBBarcodeScanner.m#L186

That is, I tried replacing that line with `return YES;` and the submission was accepted. I take it that an app that makes any UIImagePickerController call (even if it’s not actually requesting access to the user’s photo library) must include an entry for NSPhotoLibraryUsageDescription in its Info.plist. I sent an email to Apple developer support to confirm that behavior. I’ll report back if I hear anything.

If that is the case, perhaps replacing that line with the following would work:

```
return [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo] != nil;
```

The documentation for that method only says for its return value:

> The default device used to capture data of the type indicated by mediaType.

But the header says for its return value:

> The default device with the given media type, or nil if no device with that media type exists.

I’ve confirmed that simulators return nil. It looks like the most recent device that doesn’t have a camera is a third generation iPod touch and it maxes out at iOS 5.1.1 (https://en.wikipedia.org/wiki/List_of_iOS_devices) so I didn’t test this on an actual device that doesn’t have a camera.

**Comment by mikebuss** [comment]
Thanks for finding this @brblakley. That's a really unfortunate check on the part of the App Store team. I agree that your suggested change sounds like the best route to take, and will submit a fix for this in a moment under version 3.1.0.

**Comment by mikebuss** [comment]
This is now in v3.1.0. Thanks again!
